src/ctrl/PopupMenuParamWidget.h

```diff
@@ -135,11 +164,21 @@ class PopupMenuItem : public ::rack::ui::MenuItem {
         parent->text = this->text;
         ::rack::event::Change ce;
         if (parent->getParamQuantity()) {
-            float value = index;
+            float newValue = index;
+            const float oldValue =  parent->getParamQuantity()->getValue();
             if (parent->optionalIndexToValueFunction) {
-                value = parent->optionalIndexToValueFunction(index);
+                newValue = parent->optionalIndexToValueFunction(index);
             }
-            parent->getParamQuantity()->setValue(value);
+            parent->getParamQuantity()->setValue(newValue);
+     
+            // Push ParamChange history action so user may undo this change
+			::rack::history::ParamChange* h = new ::rack::history::ParamChange;
+			h->name = "change menu";
```

Comment:
This is the string that appears in the undo menu. This might not be the best choice. Maybe "change dropdown" is better?